zipkin-common/src/main/scala/com/twitter/zipkin/storage/SpanStore.scala

```diff
@@ -15,50 +15,33 @@
  */
 package com.twitter.zipkin.storage
 
-import java.nio.ByteBuffer
-
 import com.twitter.util.FuturePools._
 import com.twitter.util.{Closable, Future}
 import com.twitter.zipkin.Constants
 import com.twitter.zipkin.common.Span
+import java.nio.ByteBuffer
 
 abstract class SpanStore extends java.io.Closeable {
 
   /**
    * Get the available trace information from the storage system.
-   * Spans in trace should be sorted by the first annotation timestamp
+   * Spans in trace are sorted by the first annotation timestamp
    * in that span. First event should be first in the spans list.
    *
    * <p/> Results are sorted in order of the first span's timestamp, and contain
-   * less elements than trace IDs when corresponding traces aren't available.
+   * up to [[QueryRequest.limit]] elements.
    */
-  def getSpansByTraceIds(traceIds: Seq[Long]): Future[Seq[Seq[Span]]]
+  def getTraces(qr: QueryRequest): Future[Seq[Seq[Span]]]
 
   /**
-   * Get the trace ids for this particular service and if provided, span name.
-   * Only return maximum of limit trace ids from before the endTs.
+   * Get the available trace information from the storage system.
+   * Spans in trace are sorted by the first annotation timestamp
+   * in that span. First event should be first in the spans list.
```

Comment:
not clear on this last part - meaning that annotations will be sorted within each span?  Also, should this do a merge of spans with the same ID?  Otherwise it returns redundant data.